### PR TITLE
Samza 1186: Rename Processor to Job

### PR DESCRIPTION
**Pull request by xinyuiscool**
Now we have the top level Samza application, and each stage is called a job, the previous introduced "processor" naming should be renamed as Job. This includes renaming PrcessorGraph to JobGraph, and ProcessorNode to JobNode, etc.

### CODE REVIEW

File: samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java [hunk 1]
```diff
@@ -27,16 +27,16 @@
 
 
 /**
- * A StreamEdge connects the source {@link ProcessorNode}s to the target {@link ProcessorNode}s with a stream.
+ * A StreamEdge connects the source {@link JobNode}s to the target {@link JobNode}s with a stream.
  * If it's a sink StreamEdge, the target ProcessorNode is empty.
  * If it's a source StreamEdge, the source ProcessorNode is empty.
```
Comment (jmakes) [inline samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java]: X2
s/ProcessorNode/JobNode

File: samza-core/src/main/java/org/apache/samza/execution/JobNode.java [hunk 2]
```diff
@@ -75,22 +87,22 @@ void addOutEdge(StreamEdge out) {
...
+    configs.put(JobConfig.JOB_NAME(), jobName);
 
     List<String> inputs = inEdges.stream().map(edge -> edge.getFormattedSystemStream()).collect(Collectors.toList());
     configs.put(TaskConfig.INPUT_STREAMS(), Joiner.on(',').join(inputs));
-    log.info("Processor {} has generated configs {}", id, configs);
+    log.info("Job {} has generated configs {}", jobName, configs);
 
-    String configPrefix = String.format(CONFIG_PROCESSOR_PREFIX, id);
-    // TODO: Disallow user specifying processor inputs/outputs. This info comes strictly from the pipeline.
+    String configPrefix = String.format(CONFIG_PROCESSOR_PREFIX, jobName);
```
Comment (jmakes) [inline samza-core/src/main/java/org/apache/samza/execution/JobNode.java]: CONFIG_JOB_PREFIX and we should probably change the config constant too

File: samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java [hunk 3]
```diff
@@ -57,73 +57,75 @@ public ExecutionPlanner(Config config, StreamManager streamManager) {
     this.streamManager = streamManager;
   }
 
-  public ProcessorGraph plan(StreamGraph streamGraph) throws Exception {
-    // create physical processors based on stream graph
-    ProcessorGraph processorGraph = createProcessorGraph(streamGraph);
+  public JobGraph plan(StreamGraph streamGraph) throws Exception {
```
Comment (jmakes) [inline samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java]: Personal preference, I'd call this an ExecutionPlan. 

As in, "An execution planner produces an execution plan"

File: samza-core/src/main/java/org/apache/samza/execution/JobNode.java [hunk 4]
```diff
@@ -39,24 +39,36 @@
  * to remote cluster. In LocalExecutionEnvironment, it's a set of StreamProcessors for local execution.
  * A ProcessorNode contains the input/output, and the configs for physical execution.
```
Comment (jmakes) [inline samza-core/src/main/java/org/apache/samza/execution/JobNode.java]: s/ProcessorNode/JobNode

File: samza-core/src/test/java/org/apache/samza/execution/TestProcessorGraph.java [hunk 5]
```diff
@@ -33,10 +33,10 @@
 
 public class TestProcessorGraph {
```
Comment (jmakes) [inline samza-core/src/test/java/org/apache/samza/execution/TestProcessorGraph.java]: s/Processor/Job

File: samza-core/src/main/java/org/apache/samza/execution/JobNode.java (hunk 4 above)
Comment (xinyuiscool) [inline samza-core/src/main/java/org/apache/samza/execution/JobNode.java]: Good catch. Fixed.

File: samza-core/src/main/java/org/apache/samza/execution/JobNode.java (hunk 2 above)
Comment (xinyuiscool) [inline samza-core/src/main/java/org/apache/samza/execution/JobNode.java]: Fixed.

File: samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java (hunk 1 above)
Comment (xinyuiscool) [inline samza-core/src/main/java/org/apache/samza/execution/StreamEdge.java]: fixed.

File: samza-core/src/test/java/org/apache/samza/execution/TestProcessorGraph.java (hunk 5 above)
Comment (xinyuiscool) [inline samza-core/src/test/java/org/apache/samza/execution/TestProcessorGraph.java]: Good catch! Fixed.

File: samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java (hunk 3 above)
Comment (xinyuiscool) [inline samza-core/src/main/java/org/apache/samza/execution/ExecutionPlanner.java]: Let me be lazy and keep JobGraph for now :). If we decide to go with ExecutionPlan, I will do another renaming change.

**Pull request closed without merging**